packages/@ourworldindata/grapher/src/barCharts/DiscreteBarChart.tsx

```diff
@@ -405,11 +417,15 @@ export class DiscreteBarChart
                 {this.hasColorLegend && (
                     <HorizontalNumericColorLegend manager={this} />
                 )}
-                <HorizontalAxisComponent
-                    bounds={boundsWithoutColorLegend}
-                    axis={yAxis}
-                    preferredAxisPosition={innerBounds.bottom}
-                />
+                {/* only show the axis if all columns share a unit */}
+                {new Set(this.yColumns.map((column) => column.unit)).size ===
+                    1 && (
```

Comment:
I feel like this is too complex to be inline; you could instead extract it to `uniqueColumnUnits` or a similar `@computed` property.